packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -329,6 +333,49 @@ class EmbraceManager: NSObject {
         
     }
     
+    @objc(startView:resolver:rejecter:)
+    func startView(_ viewName: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+        let startTimeDate = Date()
+        let startTimeMS = NSNumber(value: startTimeDate.timeIntervalSince1970 * 1000)
+        
+        startSpan("emb-screen-view", parentSpanId: "", startTimeMs: startTimeMS, resolver: { spanId in
+            guard let spanId = spanId as? String else {
+                reject("START_SPAN_ERROR", "Failed to start span", nil)
+                return
+            }
+            
+            let totalAttributes = 2
+            var successfulAttributes = 0
+            
+            let attributeCompletion: (Bool) -> Void = { success in
+                if success {
+                    successfulAttributes += 1
+                }
+                if successfulAttributes == totalAttributes {
+                    resolve(spanId)
+                }
+            }
+            
+            addSpanAttributeToSpan(spanId, key: "view.name", value: viewName, resolver: { result in
+                attributeCompletion(result as? Bool ?? false)
+            }, rejecter: reject)
+            
+            addSpanAttributeToSpan(spanId, key: "emb.type", value: "ux.view", resolver: { result in
+                attributeCompletion(result as? Bool ?? false)
+            }, rejecter: reject)
+            
+        }, rejecter: reject)
+        
+    }
+    
+    @objc(endView:resolver:rejecter:)
+    func endView(_ spanId: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+        let endTimeDate = Date()
+        let endTimeMS = NSNumber(value: endTimeDate.timeIntervalSince1970 * 1000)
+        
+        stopSpan(spanId, errorCodeString: "", endTimeMs: endTimeMS, resolver: resolve, rejecter: reject)
```

Comment:
this one is more direct but I think you could copy the relevant lines from `stopSpan` since you don't care about error code, end time

Reply:
The same with start, if I do that I will rewrite the whole function